plutus-contract/src/Plutus/Contract/Test.hs

```diff
@@ -332,6 +333,18 @@ dataAtAddress address check =
               <+> foldMap (foldMap pretty . Ledger.txData . Ledger.txOutTxTx) utxo)
       pure result
 
+outputAtAddress :: forall d. FromData d => Address -> ([(d, Value)] -> Bool) -> TracePredicate
+outputAtAddress address check =
+  flip postMapM (L.generalize $ Folds.utxoAtAddress address) $ \utxo -> do
```

Comment:
It'd be nice if we could share more code between this and `dataAtAddress`.

Reply:
Since this is only for testing, we suspect this can even replace `dataAtAddress` as a generalization that covers more cases. Breaking people's CI/CD is a bit rich, but the fix should be trivial enough.

Reply:
Because this is just for testing so I think we should keep `dataAtAddress` in case others just want to check the datums?